src/OpenApi/src/Services/OpenApiDocumentService.cs

```diff
@@ -31,7 +42,34 @@ public Task<OpenApiDocument> GetOpenApiDocumentAsync()
             Paths = GetOpenApiPaths(capturedTags),
             Tags = [.. capturedTags]
         };
-        return Task.FromResult(document);
+        await ApplyTransformers(document, cancellationToken);
+        return document;
+    }
+
+    private async Task ApplyTransformers(OpenApiDocument document, CancellationToken cancellationToken = default)
+    {
+        var documentTransformerContext = new OpenApiDocumentTransformerContext
+        {
+            DocumentName = documentName,
+            ApplicationServices = serviceProvider,
+            DescriptionGroups = apiDescriptionGroupCollectionProvider.ApiDescriptionGroups.Items,
+        };
+        for (var i = 0; i < _options.DocumentTransformers.Count; i++)
+        {
+            var transformer = _options.DocumentTransformers[i];
+            // Delayed initialization for DI-activated transformers
+            // until the first time they are used when we have access
+            // to the target service provider.
+            if (transformer is ActivatedOpenApiDocumentTransformer activatedTransformer)
+            {
+                activatedTransformer.Initialize(serviceProvider);
```

Comment:
I guess this is idempotent?

Reply:
Yes, it'll create an activated instance of the given transformer type once and then cache it for future use.

Reply:
This feels sketchy. If the transformer type is scoped and uses scoped services then it will forever be instantiated with those scoped services which will result in wrong behavior for future requests.

Look at `HubFilter` for what we did for a similar concept:
https://source.dot.net/#Microsoft.AspNetCore.SignalR.Core/Internal/HubFilterFactory.cs,2de9070ddf65611c